pom.xml

```diff
@@ -1018,6 +944,83 @@
+              <execution>
+                <id>flatten-revision</id>
+              </execution>
+
+              <!-- BOMs should ideally _only_ contain dependencies for modules of the project.
+                   No invasive properties or plugin/dependency management – either from the BOM itself or its parent.
+                   The following execution exactly performs that trimming operation. -->
+              <execution>
+                <id>flatten-bom</id>
+                <goals>
+                  <goal>flatten</goal>
+                </goals>
+                <phase>process-resources</phase>
+                <configuration>
+                  <flattenMode>bom</flattenMode>
+                  <!-- POM `ElementHandling` is pretty cryptic: https://www.mojohaus.org/flatten-maven-plugin/apidocs/org/codehaus/mojo/flatten/ElementHandling.html
+                       Trial-and-error has shown that we should use either `remove` or `interpolate`.
+                       `remove` simply removes the element.
+                       `interpolate` takes the element from the original POM with variables interpolated.
+                       Avoid using `resolve`, which uses the effective POM where inherited changes from the parent are also incorporated. -->
+                  <pomElements>
+                    <build>remove</build>
+                    <dependencyManagement>interpolate</dependencyManagement>
+                    <distributionManagement>remove</distributionManagement>
+                    <profiles>remove</profiles>
+                    <properties>remove</properties>
+                    <repositories>remove</repositories>
+                    <!-- Keep the `parent`!
+                         Otherwise, this causes build failures for artifacts whose dependencies need to be resolved at runtime.
+                         Consider the following problem experienced in `log4j-changelog-maven-plugin`:
+
+                         1. `log4j-changelog-maven-plugin` depends on `log4j-changelog`
+                         2. `log4j-changelog` inherits from `log4j-tools-parent`
+                         3. `log4j-tools-parent` inherits from `log4j-tools-bom`
+                         4. `log4j-tools-bom` inherits from `logging-parent`
+                         5. `logging-parent` contains `dependencyManagement`, etc. entries that are used by `log4j-changelog-maven-plugin`
+                         6. Dependencies of `log4j-changelog-maven-plugin` et al. is resolved *at runtime*
+                         7. Though at runtime, the deployed `log4j-tools-bom` is used, which is flattened and hence doesn't have a parent!
+                         8. Hence, at runtime, `log4j-transform-maven-plugin` execution fails due to missing logic that should have been inherited from `logging-parent`
+
+                         To avoid this, parents should better be kept while flattening BOMs. -->
+                    <parent>keep</parent>
```

Comment:
Do we need this?

Our troubles were caused exactly by https://github.com/apache/logging-log4j2/issues/3066, which is fixed by this PR.

Reply:
@ppkarwasz, yes, `<parent>keep` is needed for Maven plugins. See the use case cited above this line. Do you have another idea?

Reply:
Maven plugins in this case are just simple JARs.
The problem with `log4j-changelog-maven-plugin` (see [POM file](https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-changelog-maven-plugin/0.5.0/log4j-changelog-maven-plugin-0.5.0.pom)) was that:

* it depended on `log4j-changelog` ([POM file](https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-changelog/0.5.0/log4j-changelog-0.5.0.pom)).
* `log4j-changelog` depended on `org.osgi:osgi.annotation` (unknown version).
* the version of `org.osgi:osgi.annotation` was defined in `logging-parent`, but the path to the parent was broken.

Reply:
@ppkarwasz, ah, now I see! You're right! Removed `<parent>keep` in 0885ebe23096ea09e7423b3e1fc95c9a3870a505.